plugin/nns/nns.go

```diff
@@ -72,21 +76,41 @@ func (n NNS) Transfer(zone string, serial uint32) (<-chan []dns.RR, error) {
 	return ch, nil
 }
 
+func (n *NNS) setDNSDomain(name string) {
+	n.dnsDomain = strings.Trim(name, dot)
+}
+
+func (n NNS) prepareName(name string) string {
+	if n.nnsDomain != "" {
+		name = strings.TrimSuffix(name, dot)
```

Comment:
Seems that `name = strings.TrimSuffix(name, dot)` can be executed before `if` branch and we can `return name` in the end. Didn't test it, though.